utils/rowgroup/rowaggregation.cpp

```diff
@@ -4554,19 +4570,31 @@ void RowAggregationDistinct::addRowGroup(const RowGroup* pRows,
 //------------------------------------------------------------------------------
 void RowAggregationDistinct::doDistinctAggregation()
 {
-  while (dynamic_cast<RowAggregationUM*>(fAggregator.get())->nextRowGroup())
+  fAggregator->finalAggregation();  // TODO: Perhaps move somewhere else?
```

Comment:
Not sure if that TODO should stay in. If I remember correctly I added it, because I first tried putting the `fAggregator->finalAggregation()` line into the `doThreadedAggregate()` function before `doDistinctAggregation()` gets called there, but it somehow didn't work because it wasn't able to access the correct fAggregator object in that place. 
I wasn't able to figure out why though, even though it puzzled me.

Reply:
Moved this method on top to give this idea a try.